### PR TITLE
Jeff/Add edit book functionality

### PR DESCRIPTION
**Pull request by liu-jeffrey**
## Notion ticket link

[Edit Review: Load Book Data](https://www.notion.so/uwblueprintexecs/Edit-Review-Load-Book-Data-4cb2686d67414f19817b262fae0f4758)

## Implementation description
* Create ability to edit a previously-saved book.

## Steps to test
1. Go to (http://localhost:3000/create-review)[http://localhost:3000/create-review]
2. Create and save a book
3. Click on the created book from the create-review page
4. Ensure that all data is preserved and presented properly
5. Make additional changes in the modal, click on "Save Changes"
6. Open the same book that you had edited, check that edited changes remain.

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have run docker-compose up and my project compiled
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/CreateReview/BookModal.tsx [hunk 1]
```diff
@@ -41,7 +43,14 @@ const kMaxPrice = 1000;
  * Modal for user to input new book for review
  * */
 const BookModal = (props: BookModalProps): React.ReactElement => {
-  const { isOpen, onClose, booksAdded, handleBooksAdded } = props;
+  const {
```
Comment (susheelk) [inline frontend/src/components/pages/CreateReview/BookModal.tsx:46]: Might be a good idea to create an interface for the props (with documentation), like so: https://github.com/uwblueprint/ccbc/blob/development/frontend/src/components/PreviewReviewModal.tsx#L18

File: frontend/src/components/pages/CreateReview/BookModal.tsx [hunk 2]
```diff
@@ -75,30 +84,61 @@ const BookModal = (props: BookModalProps): React.ReactElement => {
...
-    setPublicationYear("");
-  };
-
   // Calls clearBookData() whenever modal is closed
   useEffect(() => {
-    if (isOpen === false) clearBookData();
-  }, [isOpen]);
+    /** Clears book data */
+    const clearBookData = () => {
+      setCurrBook(null);
```
Comment (susheelk) [inline frontend/src/components/pages/CreateReview/BookModal.tsx:91]: Do you think it's worth having both a `currBook` and every field of the Book individually in state?

File: frontend/src/components/pages/CreateReview/BookModal.tsx [hunk 3]
```diff
@@ -75,30 +84,61 @@ const BookModal = (props: BookModalProps): React.ReactElement => {
...
+      setFormat(bookFormat.format);
+      setPrice(parseInt(bookFormat.price, 10));
+      setIsbn(bookFormat.isbn);
+
+      const bookPublisher = book.publishers[0];
+      setPublisher(bookPublisher.fullName);
+      setPublicationYear(bookPublisher.publishYear.toString());
+    };
+
+    if (isOpen === false) {
```
Comment (susheelk) [inline frontend/src/components/pages/CreateReview/BookModal.tsx:136]: (minor) `if (!isOpen)` might be cleaner

File: frontend/src/components/pages/CreateReview/BookModal.tsx [hunk 4]
```diff
@@ -165,7 +205,12 @@ const BookModal = (props: BookModalProps): React.ReactElement => {
       publishers: publisherObj,
       seriesName,
     };
-    handleBooksAdded([...booksAdded, newBook]);
+
+    if (currBook) {
+      Object.assign(currBook, newBook);
```
Comment (susheelk) [inline frontend/src/components/pages/CreateReview/BookModal.tsx:210]: Is `currBook` a reference here?

**Review by susheelk — CHANGES_REQUESTED** [review]
Code looks good! Will give this a test

File: frontend/src/components/pages/CreateReview/BookModal.tsx (hunk 4 above)
Comment (liu-jeffrey) [reply frontend/src/components/pages/CreateReview/BookModal.tsx:210]: Yessir I believe so

Comment (susheelk) [reply frontend/src/components/pages/CreateReview/BookModal.tsx:210]: Makes sense

**Comment by susheelk** [comment]
Tested and functionality looks good :fireworks: 
### Cases
* Changing Title (title changes in page) :heavy_check_mark: 
* Changing Order (order changes in page) :heavy_check_mark: 
* Removing required fields (Doesn't allow changes to be saved) :heavy_check_mark: 
* Changing other non-displayed fields :heavy_check_mark: 
* Removing/adding title :heavy_check_mark: 

Should be good to merge pending minor comments

File: frontend/src/components/pages/CreateReview/CreateReviewPage.tsx [hunk 5]
```diff
@@ -193,8 +197,10 @@ const CreateReview = (): React.ReactElement => {
...
                 index={i}
                 book={book}
-                showModal={setShowDeleteBookModal}
-                setIndex={setDeleteBookIndex}
+                showDeleteBookModal={setShowDeleteBookModal}
+                setDeleteBookIndex={setDeleteBookIndex}
+                showBookModal={onOpenBookModal}
+                setCurrBook={setCurrBook}
               />
             ))}
```
Comment (obaranek) [inline frontend/src/components/pages/CreateReview/CreateReviewPage.tsx:206]: I know this is not part of your ticket, but are we good to remove the commented code in this file?

**Review by obaranek — COMMENTED** [review]
just one very minor comment. Apart from that lgtm. Good Job!

File: frontend/src/components/pages/CreateReview/CreateReviewPage.tsx (hunk 5 above)
Comment (liu-jeffrey) [reply frontend/src/components/pages/CreateReview/CreateReviewPage.tsx:206]: Not sure, i'm concerned that some of the code could be used for tags, @susheelk?

Comment (susheelk) [reply frontend/src/components/pages/CreateReview/CreateReviewPage.tsx:201]: Resolved in call

**Review by susheelk — APPROVED** [review]
lgtm!

**Review by obaranek — APPROVED** [review]

**Pull request merged**